frontend/test/__support__/e2e/integration/visit-dashboard.cy.spec.js

```diff
@@ -0,0 +1,57 @@
+import { restore, visitDashboard } from "../cypress";
+import { USERS } from "../cypress_data";
+
+import { setup } from "./visit-dashboard";
+
+describe(`visitDashboard e2e helper`, () => {
+  Object.keys(Cypress._.omit(USERS, "sandboxed")).forEach(user => {
+    context(`${user.toUpperCase()}`, () => {
+      beforeEach(() => {
+        restore();
+        cy.signInAsAdmin();
+
+        setup();
+
+        if (user !== "admin") {
+          cy.signIn(user);
```

Comment:
BTW what happens if `admin` user is going through `cy.signIn`?

Reply:
https://github.com/metabase/metabase/pull/21801/files#diff-248e295e41a02633b63797eaff18f1d1f86ff8e82bdf82ac5a0d542846b0bc4cR11

We're already signed in as admin. No need to do it twice.
And it's crucial to sign in as admin in order to do the `setup`.